### PR TITLE
Add documentation for contact and remove unused constraint types

### PR DESCRIPTION
**Pull request by dewenyushu**
Add documentations for:

- MechanicalContactConstraint
- NodalArea
- ContactPressureAux

Remove following contact constraints:

- GluedContactConstraint
- MultiDContactConstraint
-  OneDContactConstraint
- SparsityBasedContactConstraint

Refs #14813

### CODE REVIEW

**Comment by dewenyushu** [comment]
@bwspenc PR for adding contact documentation

**Comment by permcody** [comment]
@bwspenc - we have a lot of open PRs - if you get a chance, see if you can find a few minutes to get caught up or delegate.

**Comment by bwspenc** [comment]
@permcody On it

File: modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md [hunk 1]
```diff
@@ -1 +1,57 @@
...
+
+### Contact Models
+
+The `glued` contact does not allow slip between the nodes once they come into contact, i.e., $|\boldsymbol{v}_{T}| = 0$,
+where $\boldsymbol{v}_{T}$ is the tangential relative velocity.
+ Due to the no-slip condition, the glued contact is often used to model bonded connections in e.g.,  structural analysis.
+
+In contrast to the `glued` contact, the `frictionless` contact allows slip between the nodes that are in contact. However, no tangential traction is assumed, i.e., $|\boldsymbol{v}_{T}| > 0$ and ${t}_{T} = {0}$. The frictionless contact is often utilized when the contact surface is pretty smooth thus tangential fraction along the interface can be omitted.
```
Comment (bwspenc) [inline modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md]: Drop 'However'

File: modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md [hunk 2]
```diff
@@ -1 +1,57 @@
...
+where $\mu$ is the friction coefficient.
+
+### Contact Formulations
+
+To enforce the contact constraints, three types of formulations are available. These are `kinematic`, `penalty`, and `mortar`.
+The `kinematic` and `penalty` formulations have been developed for some time and turn out to be robust for many mechanical contact problems. Recent efforts have been put in developing a mortar-based contact formulation. In this documentation, however,  we focus on `kinematic` and `penalty` formulations. For readers who are interested in mortar-based mechanical contact, please go to [mortar constraint system](Constraints/index.md) for more information.
+
+The overall workflow of imposing a mechanical constraint is as follows: 1) Determine whether penetration occurs. If so, constrain displacement; 2) In the normal direction, eliminate the penetration; 3) In the tangential direction, slip must satisfy friction model, no slit is allowed for the `glued` model, no constraint is applied for the `frictionless` model, Kuhn-Tucker conditions must be satisfied for the `coulomb` model. The mechanical contact enforcement steps for the `glued` and `frictionless` models are summarized below:
```
Comment (bwspenc) [inline modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md]: slit -> slip

Comment (bwspenc) [reply modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md]: Kuhn-Tucker conditions must be satisfied in the tangential direction for the `coulomb` model.

File: modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md [hunk 3]
```diff
@@ -1 +1,57 @@
...
+
+For `kinematic`:
+
+|  Glued  | Frictionless |
+| ---------- | ------------ |
+|  $\boldsymbol{f}_c = -\boldsymbol{r}_{s_{copy}}$     | $\boldsymbol{f}_c = -\boldsymbol{n}\boldsymbol{n}^{T} \boldsymbol{r}_{s_{copy}}$  |
+|  $\boldsymbol{r}_s = \boldsymbol{r}_s+\boldsymbol{f}_c+k_p(\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$     | $\boldsymbol{r}_s = \boldsymbol{r}_s + \boldsymbol{f}_c + \boldsymbol{n}\boldsymbol{n}^{T} k_p (\boldsymbol{x} - \boldsymbol{x}_{p_{t-1}})$ |
+|  $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$     | $\boldsymbol{r}_m = \boldsymbol{r}_m - \phi_i \boldsymbol{f}_c$ |
+
+Note the kinematic method uses penalty parameter only to penalize error. Therefore, the converged solution has no error due to penalty compliance
```
Comment (bwspenc) [inline modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md]: Note that the kinematic

Comment (bwspenc) [reply modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md]: only to penalize error -> only to enforce the condition that the slave node has the desired position on the master face, so the errors are very small in the converged solution. Unlike in penalty formulations, penalty compliance is not introduced on the contact surface.

File: modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md [hunk 4]
```diff
@@ -1 +1,57 @@
...
+
+For the frictional contact, the `penalty` method calculates normal force and tangential predictor force using penalty stiffness and check for frictional capacity. While using the `kinematic` method, the deformation of surrounding material serves the role of the penalty parameter. The problem is solved using nested solves. The normal and tangential contact tractions are computed based on strict enforcement of glued constraint. The diagonalized stiffness of slave nodes are used as regularization parameter to calculate slip.
```
Comment (bwspenc) [inline modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md]: Drop everything starting with 'The problem is solved using nested solves.' That was previously the case, but not any more.

**Review by bwspenc — CHANGES_REQUESTED** [review]
Thanks for putting this together! Just some minor editing suggestions.

File: modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md (hunk 4 above)
Comment (dewenyushu) [reply modules/contact/doc/content/source/constraints/MechanicalContactConstraint.md]: Thanks @bwspenc ! Nice catch!

**Comment by dewenyushu** [comment]
@bwspenc comments on the documentation have been addressed, thanks again! I deleted some contact constraints (docs, source files, and tests), also included in this PR.

File: modules/contact/doc/content/source/userobject/NodalArea.md [hunk 5]
```diff
@@ -1 +1,15 @@
-!template load file=stubs/moose_object.md.template name=NodalArea syntax=/UserObjects/NodalArea
\ No newline at end of file
+# NodalArea
+
+## Description
+
+<!-- The `NodalArea` computes the nodal area along the boundaries. This object is typically utilized in contact simulations while computing the contact pressure along the interface
```
Comment (bwspenc) [inline modules/contact/doc/content/source/userobject/NodalArea.md]: Did you intend to leave this here?

**Review by bwspenc — APPROVED** [review]

**Pull request merged**